Sources/StreamChat/WebSocketClient/EventMiddlewares/ChannelReadUpdaterMiddleware.swift

```diff
@@ -59,12 +61,19 @@ struct ChannelReadUpdaterMiddleware: EventMiddleware {
     private func incrementUnreadCountIfNeeded(
         for cid: ChannelId,
         message: MessagePayload,
-        session: DatabaseSession
+        session: DatabaseSession,
+        notificationCenter: EventNotificationCenter
     ) {
         guard let currentUser = session.currentUser else {
             return log.error("Current user is missing", subsystems: .webSocket)
         }
 
+        // If the message exists in the database before processing the current batch of events, it means it was
+        // already processed and we don't have to increase the unread count
+        guard notificationCenter.newMessageIds.contains(message.id) else {
```

Comment:
In line 97, we increment the unread count for this particular channel.
What this line wants to avoid is for us to increment based on a message that was already in the database before processing this event. If it was already in the database, it means it was already processed, therefore we should not be altering the unread count

Reply:
This should be the only place where we can show wrong information based on the double handling, right? All the other event middlewares process things that can't add up (e.g. typing, watching, etc).